staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go

```diff
@@ -92,3 +93,260 @@ func TestValidateLabels(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateLabelSelector(t *testing.T) {
+	successCases := []metav1.LabelSelector{
```

Comment:
Can you add names for the test cases to make them easier to review?

```go
[]struct{
    name string
    selector metav1.LabelSelector
}
```

Then use the golang subtest with the name https://blog.golang.org/subtests to make it easy to recognize what passes and what fails during runs.

Reply:
Done